### PR TITLE
[WIP] Adds in a addictive chemical called Celeritate that comes from Omega Cannabis

### PR DESCRIPTION
**Pull request by ma44**
:cl: ma44
add: A new chemical called celeritate, this chemical will speed up the body to drastic speeds, making you heal, move faster and periodically lessen the effects of slows and stuns. It will also cause brain damage as a side effect, and mutations + aging when going through the stages of addiction. Overdosing causes extreme aging, brain/clone damage as well as mutations. Don't take this chemical or else you WILL be addicted. 
add: The Omega Cannabis plant now contains celeritate in very low levels of 0.01, or 1 unit at 100 potency. 
/:cl:

TODO: Fix compiling errors, balance and test values
**No this isn't a botany change so i can powergame with "No side effect adrenaline/stimulants". The side effects aren't fun to deal with.**

### CODE REVIEW

**Comment by Kraseo** [comment]
dumb

**Comment by AdamElTablawy** [comment]
can you fucking stop

**Comment by GunHog** [comment]
Please do not put this in gaia, pick something underused instead!

**Comment by XDTM** [comment]
It's actually really weak, especially when you only get 1 unit per plant

**Comment by Kraseo** [comment]
the grammar errors trigger my inner nazi

File: code/modules/reagents/chemistry/reagents/drug_reagents.dm [hunk 1]
```diff
@@ -94,6 +94,111 @@
...
+	M.status_flags |= GOTTAGOFAST
+	if(prob(25)) //4 units on average until your healed, on average 7.2 seconds for -10 damage of the 3 basic damage types at 1.8 tick rate
+		if (M.health < 70) //No, you need some regular medicine to get to 100 health
+			M.adjustBruteLoss(-10*REM, 0)
+			M.adjustFireLoss(-10*REM, 0)
+			M.adjustToxLoss(-10*REM, 0)
+			M.adjustBrainLoss(1*REM, 0) //Brain starts to deterirate from the speed for some reason
+			M.Nutrition(-10) //Body uses nutriments to repair skin and process toxins
+			M << "<span class='notice'>You feel your wounds repairing itself!</span>"
+			user.visible_message("<span class='notice'>[user] wounds seems to nearly instantly repair itself!</span>") //Invisible healing is bad, fix with visual text
```
Comment (Kraseo) [inline code/modules/reagents/chemistry/reagents/drug_reagents.dm]: bad

File: code/modules/reagents/chemistry/reagents/drug_reagents.dm [hunk 2]
```diff
@@ -94,6 +94,111 @@
...
+/datum/reagent/drug/celeritate/on_mob_life(mob/living/M)
+	M.status_flags |= GOTTAGOFAST
+	if(prob(25)) //4 units on average until your healed, on average 7.2 seconds for -10 damage of the 3 basic damage types at 1.8 tick rate
+		if (M.health < 70) //No, you need some regular medicine to get to 100 health
+			M.adjustBruteLoss(-10*REM, 0)
+			M.adjustFireLoss(-10*REM, 0)
+			M.adjustToxLoss(-10*REM, 0)
+			M.adjustBrainLoss(1*REM, 0) //Brain starts to deterirate from the speed for some reason
+			M.Nutrition(-10) //Body uses nutriments to repair skin and process toxins
+			M << "<span class='notice'>You feel your wounds repairing itself!</span>"
```
Comment (Kraseo) [inline code/modules/reagents/chemistry/reagents/drug_reagents.dm]: no

File: code/modules/reagents/chemistry/reagents/drug_reagents.dm [hunk 3]
```diff
@@ -94,6 +94,111 @@
...
+		M.adjustStunned(-2, 0)
+		M.adjustWeakened(-2, 0)
+		M.adjustStaminaLoss(-5, 0)
+		M << "<span class='notice'>You suddenly feel a surge of energy inside of you!</span>"
+		M.visible_message("<span class='notice'>[user] looks much more energetic!</span>")
+
+/datum/reagent/drug/celeritate/overdose_process(mob/living/M)
+	M.status_flags |= GOTTAGOREALLYFAST
+	if (prob(5))
+		M << "<span class='warning'>You feel like you took to much of celeritate! </span>"
```
Comment (Kraseo) [inline code/modules/reagents/chemistry/reagents/drug_reagents.dm]: stoP

File: code/modules/reagents/chemistry/reagents/drug_reagents.dm [hunk 4]
```diff
@@ -94,6 +94,111 @@
...
+
+	if (prob(5))
+		if (M.age < 101) //Don't want people to get instakilled because reasons
+			M.age += 3
+			M << "<span class='notice'>Your body feels a lot older!.</span>"
```
Comment (Kraseo) [inline code/modules/reagents/chemistry/reagents/drug_reagents.dm]: WHY A DOT AFTER EXCLAMATION POINT

File: code/modules/reagents/chemistry/reagents/drug_reagents.dm [hunk 5]
```diff
@@ -94,6 +94,111 @@
...
+			if (prob(80))
+				M << "<span class='notice'>Your body suddenly feels just like your old self.</span>"
+				dna.remove_all_mutations() //Kills all mutations
+			if (prob(20))
+				M << "<span class='notice'>Your body feels slightly different from before.</span>"
+				/mob/living/carbon/proc/randmutg()
+
+/datum/reagent/drug/celeritate/addiction_act_stage3(mob/living/carbon/human/M)
+	if (prob(1))
+		M << "<span class='notice'>Your body feels older than it used to be</span>"
```
Comment (Kraseo) [inline code/modules/reagents/chemistry/reagents/drug_reagents.dm]: PERIOD

File: code/modules/reagents/chemistry/reagents/drug_reagents.dm [hunk 6]
```diff
@@ -94,6 +94,111 @@
...
+		M.adjustToxLoss(3)
+		M.adjustBruteLoss(3)
+		M << "<span class='warning'>You feel an awful pain inside of you, if only you could get some celeritate!</span>"
+	if (prob(5))
+		if (prob(33))
+			randmutb()
+			M << "<span class='warning'>Your body tries to adept to it's new disability!</span>"
+		if (prob(33))
+			randmutvg() //Impossible to get hulk or dwarfism this way
+			M << "<span class='notice'>Your body tries to adapt to it's new evolutionary advantage!</span>"
```
Comment (Kraseo) [inline code/modules/reagents/chemistry/reagents/drug_reagents.dm]: ITS

**Comment by Kraseo** [comment]
I REALLY HATE THESE WORD CRIMES

**Comment by ma44** [comment]
If you could point out some that would be great

**Review by Kraseo — APPROVED** [review]

File: code/modules/reagents/chemistry/reagents/drug_reagents.dm (hunk 6 above)
Comment (PKPenguin321) [inline code/modules/reagents/chemistry/reagents/drug_reagents.dm]: THE

**Comment by scoopscoop** [comment]
God bless this mess

**Comment by ExcessiveUseOfCobblestone** [comment]
i like your doggy picture @scoopscoop

File: code/modules/reagents/chemistry/reagents/drug_reagents.dm (hunk 1 above)
Comment (ExcessiveUseOfCobblestone) [inline code/modules/reagents/chemistry/reagents/drug_reagents.dm]: replace with M

File: code/modules/reagents/chemistry/reagents/drug_reagents.dm (hunk 3 above)
Comment (ExcessiveUseOfCobblestone) [inline code/modules/reagents/chemistry/reagents/drug_reagents.dm]: too much**

**Comment by ExcessiveUseOfCobblestone** [comment]
this just comes off as Meth+ to me.

File: code/modules/reagents/chemistry/reagents/drug_reagents.dm (hunk 3 above)
Comment (Iamgoofball) [inline code/modules/reagents/chemistry/reagents/drug_reagents.dm]: this is also already handled by the overdose starting

File: code/modules/reagents/chemistry/reagents/drug_reagents.dm [hunk 7]
```diff
@@ -94,6 +94,112 @@
...
+		M << "<span class='notice'>You suddenly feel a surge of energy inside of you!</span>"
+		M.visible_message("<span class='notice'>[user] looks much more energetic!</span>")
+
+/datum/reagent/drug/celeritate/overdose_process(mob/living/M)
+	M.status_flags |= GOTTAGOREALLYFAST
+	if (prob(5))
+		M << "<span class='warning'>You feel like you took to much of celeritate! </span>"
+
+	if (prob(5))
+		if (M.age < 101) //Don't want people to get instakilled from stage 4 addiction
```
Comment (Iamgoofball) [inline code/modules/reagents/chemistry/reagents/drug_reagents.dm]: i have to say, age based damage is actually really funny and creative

**Comment by Iamgoofball** [comment]
overall interesting side effects, but please have someone go through the grammer and formating

File: code/modules/reagents/chemistry/reagents/drug_reagents.dm [hunk 8]
```diff
@@ -94,6 +94,112 @@
...
+		M.adjustBrainLoss(1 * M.age / 4)
+	if (prob(5))
+		M << "<span class='notice'>Your body feels a bit weird.</span>" //I have no idea what to put here
+		M.adjustCloneLoss(1 * M.age / 4) //Your body is replicating DNA fast enough for the tiny errors to matter
+/datum/reagent/drug/celeritate/addiction_act_stage1(mob/living/carbon/human/M)
+	if (prob(10))
+		M << "<span class='notice'>You feel a slight craving for some celeritate.</span>"
+
+	if (prob(1))
+		if(!M.has_dna())
```
Comment (lordpidey) [inline code/modules/reagents/chemistry/reagents/drug_reagents.dm]: err, isn't this check the opposite of what you want?

**Comment by ma44** [comment]
Closed cause im to lazy to work on this and ill see if i get active enough to finish up on this

**Pull request closed without merging**